utilities.py

```diff
@@ -0,0 +1,31 @@
+def flat(data, parent_dir):
+    '''
+    Convert JSON data to a dictionary.
+    
+    :param list/dict data : json data.
+    :param list parent_dir: json fields. 
+    
+    :return dict: {subtest:json_value}
+    '''
+    def recursive_helper(data, parent_dir, ret):
+        if isinstance(data, list):
+            for item in data:
+                ret.update(recursive_helper(item, parent_dir, ret))
+        elif isinstance(data, dict):
+            for k, v in data.items():
+                current_dir = parent_dir.copy()
+                current_dir.append(k)
+                subtest = '.'.join(current_dir)
+                if (isinstance(v, dict) and v.values()) or isinstance(v, list):
+                    ret.update(recursive_helper(v, current_dir, ret))

```

Comment:
I am not sure if you need `and values()`, since your code will do nothing with no values anyway.  then you can replace 

    if (isinstance(v, dict) and v.values()) or isinstance(v, list):

with

    if isinstance(v, (dict, list)):